### ISSUE TITLE
Hour comparison in num2date is incorrect

### ISSUE THREAD

**Issue by penguian** [COSIMA/datetime-fortran#4]
In `num2date` in [mod_datetime.f90](https://github.com/COSIMA/datetime-fortran/blob/51-avoid-date2num-compiler-bug/src/lib/mod_datetime.f90), line 1245 has:
```
  if(num2date % hour == 60)then
```
Surely that comparison should be to 24, not 60?

**Comment by russfiedler** [comment]
Definitely incorrect code.

**Comment by aekiss** [comment]
Good catch - this is in the master at wavebitscientific - maybe raise the issue there?
https://github.com/wavebitscientific/datetime-fortran/blob/master/src/lib/mod_datetime.f90#L1238

**Comment by penguian** [comment]
Done. https://github.com/wavebitscientific/datetime-fortran/issues/64

**Comment by aekiss** [comment]
This mix of integer and real arithmetic makes me nervous too.
But I think I'm right in saying 1e3 has an exact representation as a float, so I guess that helps...
https://github.com/wavebitscientific/datetime-fortran/blob/master/src/lib/mod_datetime.f90#L1215-L1228
